snippets/tf/common.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+set -e
+
+if [ -z "${SNIPPETS_REGION}" ]; then
+  echo "SNIPPETS_REGION must be set"
+  exit -1
+fi
+
+if [ -z "${KOPS_NAME}" ]; then
+  echo "KOPS_NAME is unset, please set it or source config.sh"
+  exit 1
+fi
+
+
+SNIPPETS_TF_STATE_BUCKET="snippets-shared-tf-state"
+STATE_BUCKET_REGION="us-west-2"
+
+setup_tf_s3_state_store() {
+    echo "Creating Terraform state bucket at s3://${SNIPPETS_TF_STATE_BUCKET} (region ${STATE_BUCKET_REGION})"
+    # The following environment variables are defined in config.sh
+    aws s3 mb s3://${SNIPPETS_TF_STATE_BUCKET} --region ${STATE_BUCKET_REGION}
+
+    echo "Configuring Terraform to use an encrypted remote S3 bucket for state storage"
+    # store TF state in S3
+    terraform remote config \
+        -backend=s3 \
+        -backend-config="bucket=${SNIPPETS_TF_STATE_BUCKET}" \
+        -backend-config="key=snippets-${SNIPPETS_REGION}/terraform.tfstate" \
+        -backend-config="encrypt=1" \
+        -backend-config="region=${STATE_BUCKET_REGION}"
+
+    echo "Encryption for TF state:"
+    aws s3api head-object --bucket=$SNIPPETS_TF_STATE_BUCKET --key="snippets-${SNIPPETS_REGION}/terraform.tfstate" | jq -r .ServerSideEncryption
+}
+
+check_state_store() {
+    echo "Checking state store"
+    if aws s3 ls s3://${SNIPPETS_TF_STATE_BUCKET} > /dev/null 2>&1; then
```

Comment:
The command will return `255` exit code if the bucket does not exist and will cause the script to exit due to `set -e`. Maybe set `set +e` before executing this?

I wonder if there's a special bash instructions to make some commands not to fail.